apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/template/edit.js

```diff
@@ -126,9 +107,17 @@ const TemplateEdit = compose(
 
 		useEffect( () => {
 			// Since the Block Sidebar (`edit-post/block`) is not available for the Template block,
-			// if the sidebar is open, we force toggle to the Document Sidebar.
-			if ( isSelected && isEditorSidebarOpened ) {
-				openGeneralSidebar( 'edit-post/document' );
+			// if the sidebar is open, we force toggle to the Document Sidebar, and hide the Block button.
+			const blockSidebarButton = document.querySelector(
+				'.edit-post-sidebar__panel-tabs ul li:last-child'
+			);
+			if ( isEditorSidebarOpened && blockSidebarButton ) {
+				if ( isAnyTemplateBlockSelected ) {
+					openGeneralSidebar( 'edit-post/document' );
+					blockSidebarButton.classList.add( 'hidden' );
+					return;
+				}
+				blockSidebarButton.classList.remove( 'hidden' );
```

Comment:
Interesting do we already have a utility function `hidden` provided by core? Minor and non-blocking but I'd recommend the `is-hidden` modifier, that the plugin provides, unless this is a pretty stable feature.

Reply:
Oh and I missed this on first read but we can pass a second argument to useEffect to only run this when that prop value changes. Note that we need to pass any used state/props here as an argument to avoid stale values. This is non-blocking, but maybe give it a try. https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects

```jsx
useEffect( effectFunction, [ isEditorSidebarOpened, blockSidebarButton ] )
```

Reply:
Yup, I've tried `.is-hidden` first, but I think it's specific to some components, while [`.hidden` is wp-admin-wide](https://github.com/WordPress/WordPress/blob/9793494a2c56029fc92be37198568167c57df947/wp-admin/css/common.css#L166-L175).

Reply:
@gwwar TIL the `useEffect` dependencies! Very cool! (added in 21a0b36)